app/models/concerns/password_validatable.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module PasswordValidatable
+  extend ActiveSupport::Concern
+
+  PWD_REGEX = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).*$/
+
+  private
+
+  def password_complexity
+    return if valid_pwd?
+
+    err_mess = I18n.t('devise.registrations.password_complexity_error')
+    errors.add(:password, err_mess)
+  end
+
+  def valid_pwd?
```

Comment:
Opinion: Write full words instead of short forms
```suggestion
  def valid_password?
```

Reply:
There is already a method `valid_password?` in devise, which conflicts with this method, I will change other suggested method and variable names.

Reply:
I would then use a different method name, and a comment explaining the method

Reply:
Especially if there is already a `valid_password` method in devise, it's going to be easy to get these two confused